Sources/KombuchaLib/SnapConfiguration.swift

```diff
@@ -98,13 +106,25 @@ extension SnapConfiguration.Request: Decodable {
     }
 }
 
-public struct GETSnap: Decodable {
+public struct RESTSnap: Decodable {
+    
+    public enum HTTPMethod: String, Decodable {
+        case get = "GET"
+        case post = "POST"
+        case put = "PUT"
+        case delete = "DELETE"
+        case patch = "PATCH"
+    }
+    
     public var host: String
     public var path: String
     public var queryItems: [String: String?]
     public var scheme: String
+    public var httpMethod: HTTPMethod
+    public var httpHeaders: [String: String?]
```

Comment:
so are we always requiring headers to be set? I could see someone preferring not to set headers if they don't need it. Can we set a default value here instead of requiring it in the config?

Reply:
I made it optional :)